src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -234,18 +232,27 @@ def _adjust_batch_size(
     """
     model = trainer.lightning_module
     batch_size = lightning_getattr(model, batch_arg_name)
-    new_size = value if value is not None else int(batch_size * factor)
+    if value is not None:
+        new_size = value
+    else:
+        if not isinstance(batch_size, int):
+            raise ValueError(f"Batch size attribute in LightningModule must be an integer, got: {batch_size!r}")
+        new_size = int(batch_size * factor)
+
     if desc:
         log.info(f"Batch size {batch_size} {desc}, trying batch size {new_size}")
 
     if not _is_valid_batch_size(new_size, trainer.train_dataloader, trainer):
+        assert trainer.train_dataloader is not None
         new_size = min(new_size, len(trainer.train_dataloader.dataset))
 
     changed = new_size != batch_size
     lightning_setattr(model, batch_arg_name, new_size)
     return new_size, changed
 
 
-def _is_valid_batch_size(batch_size: int, dataloader: DataLoader, trainer: "pl.Trainer"):
+def _is_valid_batch_size(batch_size: int, dataloader: Optional[Any], trainer: "pl.Trainer") -> bool:
+    if dataloader is None:
+        return True
```

Comment:
this is not the right way. there should be a dataloader to check the initial value. Fixed it #11089

Reply:
Thanks for your suggestion. I got it.